confChamber.py

```diff
@@ -87,5 +88,20 @@
     except Exception as e:
         print '%s does not seem to exist'%options.filename
         print e
-print 'Chamber Configured'
 
+if options.dictConfig:
+    try:
+        print "Configuring VFATs with dictionary values"
+
+        for x in range (len(chamber_defaults)):
```

Comment:
This is a mistake, confChamber.py is called once per detector either by the user individually or by confAllChambers.py.  Each execution corresponds to one detector.

Here you're looping over all the elements of the chamber_defaults dictionary.  So you'll be writing these registers `len(chamber_defaults)` times and the register values that are stored will correspond to the last element `len(chamber_defaults)-1` rather than the link given by the `-g` option.

Please revise, also taking into account the suggested redesign above (e.g. if a link isn't present in the dictionary it should default to those values stored in vfat_user_functions.py)